Assets/Scripts/Grid.cs

```diff
@@ -119,11 +119,13 @@ public TileData.TileType GetTileType(int row, int col)
 
     public bool CanBlockFit(int row, int col, Block block)
     {
-        //Assume each tile is 1x1 size.
+        //The block with regular tiles at specific indices can fit
+        //into the grid if and only if the corresponding indices does
+        //not have any regular and vestige tiles.
         for (int c = 0; c < block.GetWidth(); c++){
             for (int r = 0; r < block.GetHeight(); r++){
-                if (tiles[row + r, col + c].GetIsOccupied() &&
-                    block.GetIsOccupied(r, c))
+                if (tiles[row + r, col + c].GetTileType() == TileData.TileType.Regular ||
+                    tiles[row + r, col + c].GetTileType() == TileData.TileType.Vestige)
                     return false;

```

Comment:
The block.GetIsOccupied check was necessary here since we're comparing the tiles of the Block to the tiles of the Grid to check for any overlapping tiles. If the tile of the block is occupied AND the same tile of the Grid is occupied, there is overlap, and so we cannot place the tile there. I do believe the original code did that just perfectly.

Also, instead of listing every tile type that counts as "occupied" (Regular, Vestige, etc), just use GetIsOccupied. It is far more convenient. In the future, we will likely be adding a lot more "occupied" tile types, and having to expand this if statement each time we add something new would be far too inconvenient.

I do believe the original code was fine here. The actual change that needs to be made is the removal of the Vacant tiles from the generated blocks in BlockSpawner. I will handle that myself.